### PR TITLE
add the ability to specify K8S/OCP credentials on a Job Template

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/ansible/awx/issues/5735

### CODE REVIEW

**Comment by ryanpetrello** [comment]
```
bash-4.4$ cat /var/lib/awx/projects/k8s/example.yml
---

- hosts: all
  gather_facts: false
  tasks:
    - name: Get a list of all pods from any namespace
      k8s_info:
        kind: Pod
```

image

image

**Comment by wenottingham** [comment]
cc @geerlingguy - does the community.kubernetes collection take these same parameters?

**Review by jakemcdermott — APPROVED** [review]

**Comment by AlanCoding** [comment]
I'm guessing that these are expected to work specifically because of the lookup plugin

https://github.com/ansible-collections/community.kubernetes/blob/6b28ef3c17449fdd34d0b51c3846278db80239c7/plugins/connection/kubectl.py#L130

which is a very very important plugin in the grand scheme of things...

File: awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ function multiCredentialModalController(GetBasePath, qs, MultiCredentialService)
 
         scope.credentialTypes.forEach((credentialType => {
             if(credentialType.kind
-                .match(/^(machine|cloud|net|ssh|vault)$/)) {
+                .match(/^(machine|cloud|net|ssh|vault|kubernetes)$/)) {
```
Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: I'd check myself but `dnf` download speed drops to 40 kb/s while building the image locally and don't want to tie up my computer for 45 minutes:

Screenshot from 2020-07-16 08-24-17

... so I'll just ask: Is the credential type displayed on the job launch credential prompt as well? If not we'll need to update some more code & you'd want to start here: https://github.com/ansible/awx/blob/devel/awx/ui/client/src/templates/prompt/steps/credential/prompt-credential.controller.js


We'll also want to check the form, job details, and prompt on the new UI.

Comment (jakemcdermott) [reply awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: ALRIGHT!!, I made it, finally :confetti_ball:

File: awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ function multiCredentialModalController(GetBasePath, qs, MultiCredentialService)
...
                     scope.displayedCredentialTypes.push(credentialType);
             }
         }));
```
Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:117]: Screenshot from 2020-07-16 09-16-03

Icon missing on form, you can add it [here](https://github.com/ansible/awx/blob/devel/awx/ui/client/src/templates/job_templates/multi-credential/multi-credential.partial.html#L22-L37)

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

File: awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js (hunk 1 above)
Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: Icon is missing for prompts
Screenshot from 2020-07-16 09-21-30

That's rendered [here](https://github.com/ansible/awx/blob/devel/awx/ui/client/src/templates/prompt/steps/credential/prompt-credential.partial.html#L14) using `promptData.prompts.credentials`, which I _think_ ultimately comes from [here](https://github.com/ansible/awx/blob/devel/awx/ui/client/lib/components/relaunchButton/relaunchButton.component.js#L87)

Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: We'll need to make sure the new option gets displayed on the multicredential select lookup in ui_next:
cred_form

... which I presume to be something to be added [here](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/components/Lookup/CredentialLookup.jsx#L81)

Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: We'll also need to make sure the new category is displayed on launch prompts in `ui_next`:
cred_prompt

That code lives [here](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/components/LaunchPrompt/steps/CredentialsStep.jsx#L119)

Comment (jakemcdermott) [inline awx/ui/client/src/templates/job_templates/multi-credential/multi-credential-modal.directive.js:114]: JDetails for `ui_next` looks fine:
cred_jdetails

**Comment by geerlingguy** [comment]
@wenottingham - Yes `K8S_AUTH_TOKEN` and `K8S_AUTH_API_KEY` work with the K8s modules. Note that I typically prefer using a `K8S_AUTH_KUBECONFIG` instead, but it depends on how the user can authenticate to the cluster and I know with at least some versions of OCP the credentials are structured differently than I'm used to with vanilla K8s clusters. cc @AlanCoding

**Comment by ryanpetrello** [comment]
@jakemcdermott here's my latest upload w/ the old UI:

Kapture 2020-07-16 at 10 57 24

...and here's the new UI:

Kapture 2020-07-16 at 11 24 22

**Comment by ryanpetrello** [comment]
@jakemcdermott it _looks like_ the new UI doesn't currently have the validation support for "hey, you removed a required credential at prompt on launch, and you should replace it before continuing".  Do you see evidence otherwise?

```shell
~/dev/awx ack "must be included" awx/ui/client/features
awx/ui/client/features/templates/templates.strings.js
75:        CREDENTIAL_TYPE_MISSING: typeLabel => t.s('This job template has a default {{typeLabel}} credential which must be included or replaced before proceeding.', { typeLabel })
~/dev/awx ack "must be included" awx/ui_next
```

cc @keithjgrant @AlexSCorey

**Comment by jakemcdermott** [comment]
@ryanpetrello 
beautiful

**Review by jakemcdermott — APPROVED** [review]

**Comment by ryanpetrello** [comment]
@jakemcdermott I'm gonna add some tests, then will merge.

**Comment by jakemcdermott** [comment]
doit

**Review by jakemcdermott — APPROVED** [review]

**Comment by jakemcdermott** [comment]
doitt

**Review by elyezer — APPROVED** [review]

**Pull request merged**